docs/architecture/adr-xxx-proposer-slashing.md

```diff
@@ -0,0 +1,49 @@
+# ADR Proposer slashing for resource misuse attacks.
+
+## Changelog
+
+
+## Abstract
+
+Block Proposers can construct blocks at their discretion but they are free to fill blocks with invalid transactions or exceed the block gas limit. This
+
+
+## Context
+
+During the Terra blockchain failure incident in May of 2022, we observed a very large number of invalid and duplicated transactions landing on chain. This provided the first concrete evidence that on a large Tendermint chain validators were running non-standard software to propose blocks. Protocols built with the cosmos sdk need to have slashing behavior that punish validators for poorly designed block construction software.
+
+
+## Decision
+
+
+Rational/Malicious proposers may commit protocol violations when using non standard mempools to generate transaction. 
+
+There are few known situations.
+
+1. Submitting transactions with invalid sequence numbers
+2. Submitting transactions with invalid signatures
+3. Creating a block where the total of Gas_wanted > max gas for a block.
```

Comment:
I think we also  could add:
* including malformed transaction
* wrong chain_id etc...